jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -2519,33 +2522,60 @@ ecma_regexp_replace_helper_fast (ecma_replace_context_t *ctx_p, /**<replace cont
   const lit_utf8_byte_t *matched_p = NULL;
   const lit_utf8_byte_t *current_p = ctx_p->string_p;
   const lit_utf8_byte_t *last_append_p = current_p;
-  JERRY_ASSERT (ctx_p->index <= string_length);
+  ecma_length_t index = 0;
+
+  ecma_regexp_ctx_t re_ctx;
+  ecma_regexp_initialize_context (&re_ctx,
+                                  bc_p,
+                                  ctx_p->string_p,
+                                  string_end_p);
 
 #if ENABLED (JERRY_ESNEXT)
   /* Global matches always start at index 0, but Sticky matches may have a non-zero lastIndex. */
-  if (ctx_p->index > 0)
+  if ((ctx_p->flags & RE_FLAG_GLOBAL) == 0 && (re_ctx.flags & RE_FLAG_STICKY) != 0)
```

Comment:
How about `(ctx_p->flags & (RE_FLAG_GLOBAL | RE_FLAG_STICKY)) ==  RE_FLAG_GLOBAL`?

Reply:
The two flags come from different contexts. The global and unicode flags come from the replace context, and are coerced from the global and unicode poperties of the 'this' object, the rest of the flags should be used based on the [[OriginalFlags]] internal slot of the regexp instance.

Reply:
Indeed, I've missed that part.